### PR TITLE
CodePeak 2025 : Supply Chain Demand Forecasting

### PR DESCRIPTION
**Pull request by Mikeyzgoat**
## Related Issue

- Demand Forecasting with LSTM-Transformer Model

- [x] CodePeak 2025 Participant
- [ ] Contributor

Closes: #648  that will be closed through this PR

### Describe the add-ons or changes you've made

Added a demand forecasting model using LSTM-Transformer to predict future sales quantities based on historical e-commerce data. Implemented data preprocessing, time series forecasting model, and visualization of results. Fine-tuned hyperparameters and applied a smoothing technique for better trend visualization.

## Type of change

What sort of change have you made:
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- Trained the LSTM-Transformer model for 500/1000 epochs with different learning rates.
- Validated results by comparing predicted vs. actual sales quantities.
- Visualized predictions using Matplotlib.
- Applied rolling forecasts for future predictions.

## Checklist:
- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

training with epoch 500 learning rate 0.01
> Training with epoch 1000 and learning rate 0.0001
training with epoch 1000 learning rate 0.0001
> Training with epoch 1000 and learning rate 0.005
training with epoch 1000 learning 0.005

### CODE REVIEW

**Comment by prathimacode-hub** [comment]
Hola, @Mikeyzgoat. Kindly use 3-4 algorithms and compare the models and accuracy. Also involve the lifecycle of data science : data analysis, data visualization, 3-4 models, compare models with accuracy and score followed by a summary statement mentioning best fit model out of implemented algorithms. If possible, add a .pkl file for best fit model. As of now, u had incorporated only LSTM model. Do the changes accordingly. Thank you.

**Comment by Mikeyzgoat** [comment]
Sure thing @prathimacode-hub , I will add SARIMAX and XGBOOST for comparision, prior to this I ll also run data cleaning, preprocessing and enhance feature extraction. Thanks for the review!, I will reach out within 1-2 days.

**Comment by prathimacode-hub** [comment]
> Sure thing @prathimacode-hub , I will add SARIMAX and XGBOOST for comparision, prior to this I ll also run data cleaning, preprocessing and enhance feature extraction. Thanks for the review!, I will reach out within 1-2 days.

Perfect, go ahead..

**Comment by prathimacode-hub** [comment]
Any update? @Mikeyzgoat

**Comment by Mikeyzgoat** [comment]
> Any update? @Mikeyzgoat

Yes, i might need a day or two, was caught up in some office work. Will revert back ASAP

**Review by prathimacode-hub — APPROVED** [review]
Your PR is approved. Well done.. 🙌 @Mikeyzgoat

**Comment by Mikeyzgoat** [comment]
Thank you @prathimacode-hub , can you update the same on CODEPEAK please, thanks again!. Happy coding

**Comment by prathimacode-hub** [comment]
> Thank you @prathimacode-hub , can you update the same on CODEPEAK please, thanks again!. Happy coding

Sure, I asked for process to add tags, so shall do at earliest.. @Mikeyzgoat
